app/controllers/users_controller.rb

```diff
@@ -0,0 +1,29 @@
+class UsersController < ApplicationController
+  def show
+    @user = User.find_by id: params[:id]
+    return if @user
+
+    flash[:warning] = "Not found user!"
+    redirect_to root_path
+  end
```

Comment:
```suggestion
  def show; end
```
Đưa các code bên trong ra thành hàm load_user và dùng trong before_action nha